astropy/utils/masked/tests/test_masked.py

```diff
@@ -661,6 +661,15 @@ def test_setitem_np_ma_masked(self, item):
         assert_array_equal(base.unmasked, self.a)
         assert_array_equal(base.mask, expected_mask)
 
+    @pytest.mark.parametrize("item", VARIOUS_ITEMS)
+    def test_hash(self, item):
+        ma_part = self.ma[item]
+        if ma_part.shape or ma_part.mask.any():
```

Comment:
I'm having a hard time following this test code given the inheritance stack, maybe we can make it slightly more readable
```suggestion
        if ma_part.ndim > 0 or ma_part.mask.any():
```

Reply:
Done